lib/data_update_scripts/20220209110134_reprocess_user_subscription_liquid_tag_posts.rb

```diff
@@ -0,0 +1,17 @@
+module DataUpdateScripts
```

Comment:
@jeremyf can I tag you in for some help here? I've had a first crack at it - it appears to work (at least in the current test cases), but I'd definitely appreciate some keen eyes on it.

I had a couple of questions in particular:

- Should I have a catch block for any md that can't be reprocessed, and if so... what to do with the error?
- It is possible (maybe not super likely) that a user had permission to use this liquid tag when they originally posted, but no longer have that permission. At the minute I'm only re-processing the HTML if the user _still_ has permission because otherwise, the app throws an error when processing. But actually, I think in an ideal world we'd want to skip that authorisation check in this instance. But how we would do that is beyond me 😅

Reply:
@aitchiss I'll definitely take a crack at this.  In regards to catching the error, I look to @jgaskins to provide some guidance.

Reply:
@jeremyf We may very well run into errors on this for multiple reasons. Apart from issues in reprocessing the markdown, the query takes a full minute because `body_markdown` doesn't have an index that a `LIKE` query can use. But IIRC DEV's statement timeout is 30 seconds for DUSs.

On the bright side, this query only matches 0.00062% of published articles on DEV  (over 99.999% of articles do _not_ use it) and all of them were published over 18 months ago. I don't know how many other Forem instances use this, but if it's proportional to DEV's volume it's probably 0-1 on most Forems. @aitchiss @jeremyf With that in mind, how terrible would it be if we skipped the DUS altogether?

Reply:
This is really interesting context - thank you! When I was fixing up this liquid tag I noticed all the messaging was hard-coded to DEV, so I think it's fairly likely this was never used in other Forems. Given the nature of the tag I would also hazard guess the vast majority of uses were by our team.

> With that in mind, how terrible would it be if we skipped the DUS altogether?

I'm starting to think it wouldn't be terrible 😄 I'll check with Gracie because I think she's probably most familiar with where we've been using this tag

Reply:
@jgaskins - don't suppose you can send me a list of the posts that match that query (author, publish date, url, title)?

Please just tell me if that's a massive PITA 😆

Reply:
@aitchiss lol Nope, that's easy. I'll get those over to you.